### PR TITLE
Fix ability to set Success and Failure in non advanced audit policies

### LINKED ISSUE #60142: [BUG] win_lpgo has problems with values containing comma
**Description**
It appears that the win_lgpo state module gets confused if a security setting value contains a comma. For instance, property "Audit Logon Events" fully on should be "Success, Failure". However, under some state files, trying to set that value will result in it being set to "Success" only. 

The win_lgpo command line module works fine; it seems to be an issue with using the *state* module. Also, if I create a state with only 1 security setting, it seems to work. I see the problem when the setting is part of a bigger list, so I am guessing it is a parsing problem on bulk results (my debugging seemed to show that Salt fetches blocks of GPO entries).

**Setup**
This policy file contains a small set (commented out) which does work, and a big set which produces the bad results.
```
# update the group policy right after changing settings
group-policy-update:
  cmd.wait:
    - name: 'gpupdate /force'

# good_policy:
#   lgpo.set:
#     - computer_policy:
#         Audit Logon Events: "Success, Failure"
#     - watch_in:
#       - cmd: group-policy-update

bad_policy:
  lgpo.set:
    - computer_policy:
        Audit Credential Validation: Success and Failure
        Audit Account Management: 'Success, Failure'
        Audit Other Account Management Events: Success
        Audit User Account Management: Success and Failure
        Audit Security Group Management: Success and Failure
        Audit Account Lockout: Success
        Audit Logoff: Success
        Audit Logon Events: 'Success, Failure'
        Audit Logon: Success and Failure
        Audit Special Logon: Success
        Audit Audit Policy Change: Success and Failure
        Audit Sensitive Privilege Use: Success and Failure
        Audit Authentication Policy Change: Success and Failure
    - watch_in:
      - cmd: group-policy-update

```

**Steps to Reproduce the behavior**
Output from running "bad policy"
```
MyMachine:
----------
          ID: bad_policy
    Function: lgpo.set
      Result: True
     Comment: The following policies changed:
              Audit Logon Events
     Started: 14:43:37.369994
    Duration: 3797.638 ms
     Changes:   
              ----------
              new:
                  ----------
                  Computer Configuration:
                      ----------
                      Audit Account Management:
                          Success
                      Audit Logon Events:
                          Success
              old:
                  ----------
                  Computer Configuration:
                      ----------
                      Audit Account Management:
                          Success, Failure
                      Audit Logon Events:
                          Failure

```

**Expected behavior**
In the above case, it should have set both policies to "Success, Failure", but they ended up as just "Success".

**Screenshots**

**Versions Report**
<details><summary>salt --versions-report</summary>
(Provided by running salt --versions-report. Please also mention any differences in master/minion versions.) 

```
MyMachine:
    Salt Version:
              Salt: 3002.6
     
    Dependency Versions:
              cffi: 1.12.2
          cherrypy: 17.4.1
          dateutil: 2.8.0
         docker-py: Not Installed
             gitdb: 2.0.5
         gitpython: 2.1.10
            Jinja2: 2.10.1
           libgit2: Not Installed
          M2Crypto: Not Installed
              Mako: 1.0.7
           msgpack: 1.0.0
      msgpack-pure: Not Installed
      mysql-python: Not Installed
         pycparser: 2.19
          pycrypto: Not Installed
      pycryptodome: 3.9.8
            pygit2: Not Installed
            Python: 3.7.4 (tags/v3.7.4:e09359112e, Jul  8 2019, 20:34:20) [MSC v.1916 64 bit (AMD64)]
      python-gnupg: 0.4.4
            PyYAML: 5.3.1
             PyZMQ: 18.0.1
             smmap: 2.0.5
           timelib: 0.2.4
           Tornado: 4.5.3
               ZMQ: 4.3.1
     
    System Versions:
              dist:   
            locale: cp1252
           machine: AMD64
           release: 10
            system: Windows
           version: 10 10.0.19041 SP0

```
</details>

**Additional context**
Happy to run any tests.

### PR DESCRIPTION
**Pull request by twangboy**
### What does this PR do?
Standardizes the input for setting Success and Failure for all auditing policies. The advanced audit policies used `Success and Failure` where the normal auditing policies used `Success, Failure`. The comma was causing some problem with some of the policies so we standardized on `Success and Failure`.

### What issues does this PR fix or reference?
Fixes: https://github.com/saltstack/salt/issues/60142

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [x] Changelog - https://docs.saltstack.com/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
Yes

### CODE REVIEW

File: salt/modules/win_lgpo.py [hunk 1]
```diff
@@ -336,7 +336,7 @@ def __init__(self):
             0: "No auditing",
             1: "Success",
             2: "Failure",
-            3: "Success, Failure",
+            3: "Success and Failure",
```
Comment (Ch3LL) [inline salt/modules/win_lgpo.py:339]: Seems this would require users to update their states. Can we prevent that from occurring?

**Review by Ch3LL — CHANGES_REQUESTED** [review]

File: tests/pytests/functional/modules/win_lgpo/test_audit_settings_module.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+def test_success(enable_legacy_auditing, clear_policy):
+    _test_auditing("Success")
+
+
+def test_failure(enable_legacy_auditing, clear_policy):
+    _test_auditing("Failure")
+
+
+def test_success_and_failure(enable_legacy_auditing, clear_policy):
+    _test_auditing("Success, Failure")
```
Comment (Ch3LL) [inline tests/pytests/functional/modules/win_lgpo/test_audit_settings_module.py:112]: Can we also test `"Success and Failure"`

File: tests/pytests/functional/states/win_lgpo/test_audit_settings_state.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+def test_success(enable_legacy_auditing, clear_policy):
+    _test_auditing("Success")
+
+
+def test_failure(enable_legacy_auditing, clear_policy):
+    _test_auditing("Failure")
+
+
+def test_success_and_failure(enable_legacy_auditing, clear_policy):
+    _test_auditing("Success, Failure")
```
Comment (Ch3LL) [inline tests/pytests/functional/states/win_lgpo/test_audit_settings_state.py:124]: same here, can we test for `"Success and Failure"`

**Review by Ch3LL — CHANGES_REQUESTED** [review]

File: salt/modules/win_lgpo.py [hunk 4]
```diff
@@ -5056,42 +5056,67 @@ def _get_advaudit_defaults(option=None):
...
+    if not __salt__["file.file_exists"](f_audit):
+        if __salt__["file.file_exists"](f_audit_gpo):
+            # If the GPO audit.csv exists, we'll use that one
+            __salt__["file.copy"](f_audit_gpo, f_audit)
+        else:
+            field_names = _get_advaudit_defaults("fieldnames")
+            # If the file doesn't exist anywhere, create it with default
+            # fieldnames
+            __salt__["file.makedirs"](f_audit)
+            __salt__["file.write"](f_audit, ",".join(field_names))
```
[the remark this answers is not in the document]
Comment (twangboy) [reply salt/modules/win_lgpo.py:5093]: Oui

File: tests/pytests/functional/modules/win_lgpo/test_audit_settings_module.py (hunk 2 above)
Comment (twangboy) [reply tests/pytests/functional/modules/win_lgpo/test_audit_settings_module.py:112]: **Advanced** audit settings are set with `Success and Failure"`. **Legacy** audit settings are set with `"Success, Failure"`

File: tests/pytests/functional/states/win_lgpo/test_audit_settings_state.py (hunk 3 above)
Comment (twangboy) [reply tests/pytests/functional/states/win_lgpo/test_audit_settings_state.py:124]: Same here

**Review by Ch3LL — APPROVED** [review]

**Comment by Ch3LL** [comment]
Can we get your review here @xeacott

**Review by xeacott — APPROVED** [review]

**Pull request merged**